### PR TITLE
chore: clean up mutate function into helpers mixin

### PR DESCRIPTION
**Pull request by eschutho**
### SUMMARY
I need to make some changes to the mutator, so I'm starting off by removing some duplicate code and writing tests. No functional changes. 

### TESTING INSTRUCTIONS
Tests added

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/models/helpers.py [hunk 1]
```diff
@@ -875,7 +875,9 @@ def mutate_query_from_config(self, sql: str) -> str:
         """Apply config's SQL_QUERY_MUTATOR
 
         Typically adds comments to the query with context"""
-        if sql_query_mutator := config["SQL_QUERY_MUTATOR"]:
+        if (sql_query_mutator := config.get("SQL_QUERY_MUTATOR")) and not config.get(
+            "MUTATE_AFTER_SPLIT"
+        ):
```
Comment (eschutho) [inline superset/models/helpers.py:880]: This was pulled in from superset/connectors/sqla/models.py

**Review by hughhhh — APPROVED** [review]

File: superset/models/helpers.py [hunk 2]
```diff
@@ -875,7 +875,9 @@ def mutate_query_from_config(self, sql: str) -> str:
         """Apply config's SQL_QUERY_MUTATOR
 
         Typically adds comments to the query with context"""
-        if sql_query_mutator := config["SQL_QUERY_MUTATOR"]:
+        if (sql_query_mutator := config.get("SQL_QUERY_MUTATOR")) and not config.get(
```
Comment (john-bodley) [inline superset/models/helpers.py:878]:
```suggestion
        if (sql_query_mutator := config["SQL_QUERY_MUTATOR"]) and not config["MUTATE_AFTER_SPLIT"]):
```

File: superset/models/helpers.py [hunk 3]
```diff
@@ -875,7 +875,9 @@ def mutate_query_from_config(self, sql: str) -> str:
...
+            "MUTATE_AFTER_SPLIT"
```
Comment (john-bodley) [inline superset/models/helpers.py:879]:
```suggestion
```

File: superset/models/helpers.py (hunk 1 above)
Comment (john-bodley) [inline superset/models/helpers.py:880]:
```suggestion
```

File: superset/models/helpers.py (hunk 2 above)
Comment (john-bodley) [reply superset/models/helpers.py:878]: @eschutho both these keys are defined in `config.py` and thus there's no need to use `.get()`.

Comment (eschutho) [reply superset/models/helpers.py]: Thank you!

File: superset/models/helpers.py (hunk 3 above)
Comment (eschutho) [reply superset/models/helpers.py:884]: @john-bodley I'm circling back to this old PR to see if I can get it merged. Do you think this statement needs to be removed? I copied it over from `superset/connectors/sqla/models.py` when drying up this code.

**Pull request closed without merging**